File: Src/java/cql-to-elm/src/jvmTest/java/org/cqframework/cql/cql2elm/elm/ElmEditorTest.java
package org.cqframework.cql.cql2elm.elm;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertThrows;

import java.util.List;
import org.hl7.elm.r1.ChoiceTypeSpecifier;
import org.hl7.elm.r1.Library;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

class ElmEditorTest {
    private int editCount = 0;
    private final ElmEditor editor = new ElmEditor(List.of(element -> editCount++));

    @BeforeEach
    void beforeEach() {
        editCount = 0;
    }

    @Test
    void edit() {
        editor.edit(new Library());
        assertEquals(1, editCount);
    }

    @Test
    void applyEdits1() {
        assertThrows(NullPointerException.class, () -> editor.applyEdits(null));
    }

    @Test
    void applyEdits2() {
        editor.applyEdits(new ChoiceTypeSpecifier());
        assertEquals(1, editCount);
    }
}

File: cql-to-elm/src/jvmTest/kotlin/org/cqframework/cql/cql2elm/elm/ElmEditorTest.kt
package org.cqframework.cql.cql2elm.elm

import kotlin.collections.listOf
import org.hl7.elm.r1.ChoiceTypeSpecifier
import org.hl7.elm.r1.Element
import org.hl7.elm.r1.Library
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

internal class ElmEditorTest {
    private var editCount = 0
    private val edit: IElmEdit = IElmEdit { _: Element -> editCount++ }
    private val editor = ElmEditor(listOf(edit))

    @BeforeEach
    fun beforeEach() {
        editCount = 0
    }

    @Test
    fun edit() {
        editor.edit(Library())
        Assertions.assertEquals(1, editCount)
    }

    @Test
    fun applyEdits2() {
        editor.applyEdits(ChoiceTypeSpecifier())
        Assertions.assertEquals(1, editCount)
    }
}
